docs/docs/creating-prefixed-404-pages-for-different-languages.md

```diff
@@ -56,6 +58,96 @@ exports.onCreatePage = async ({ page, actions }) => {
 }
 ```
 
+## Use createPages API
+
+If you're using [gatsby-plugin-i18n](/docs/localization-i18n/#gatsby-plugin-i18n) to make a i18n site, and want to generate all pages dynamically by defining `markdown` files in different languages including `404.js`. Using the [`createPages`](/docs/node-apis/#createPages) API is properly a better way.
+
+First of all, create a 404 template file in `src/templates/404.js`.
+
+```jsx:title=src/templates/404.js
+import React from "react";
+import { graphql, Link } from "gatsby";
+import Layout from "../../components/layout"
+
+export const query = graphql`
+  query NotFoundQuery($langKey: String!) {
+    markdownRemark(fields: { langKey: { eq: $langKey }, fileName: { regex: "/404/" } }) {
```

Comment:
@gillkyle 
First of all, this approach uses `gatsby-plugin-i18n` only in this graphql's filter and which is avoidable by filtering filename in render method without the plugin. 

But I still think that this approach is a little bit complicated for simple 404 pages.

As I described in #23776, what I were trying to do is using `createPages` API to create prefixed 404 pages but not `onCreatePage`.
So I searched `404` in docs and ran into [the documentation page](https://www.gatsbyjs.org/docs/creating-prefixed-404-pages-for-different-languages/). But unfortunately after I read the documentation, it seems that we can only do it with `onCreatePage` API like this.
```js
page.matchPath = `/${langCode}/*`
```

I don't want to use `onCreatePage` API as it never existed in my gatsby-node.js, so I tried a lot and finally found that if I pass `matchPath` to `createPage` all things work like a charm. I think someone should face the same problem of `how to use createPages to make prefixed pages?` no matter they're using i18n plugin or not. So I opened this issue here.

What about adding a simple `hint` for those people as my first commit did?
```
You can also do it by using the [`createPages`](/docs/node-apis/#createPages) API in your project's `gatsby-node.js` file to generate prefixed 404 pages dynamically and pass `langCode` to `matchPath`.

createPage({
    path: `/${langCode}/404/`,
    matchPath: `/${langCode}/*`,
    component: page404,
    context: {
      langCode,
    },
});
```